### PR TITLE
fix: improve VTK mapper and scalar bar serializers

### PR DESCRIPTION
**Pull request by banesullivan**

### CODE REVIEW

File: trame_vtk/modules/vtk/addon_serializer.py [hunk 1]
```diff
@@ -64,6 +65,213 @@ def extractRequiredFields(
...
+    # axisTitlePixelOffset = actor.GetTextPad()
+    # position = actor.GetPosition()
+    # position2 = actor.GetPosition2()
+    # print(f'axisTitlePixelOffset: {axisTitlePixelOffset}')
+    # print(f'axisLabel: {axisLabel}')
+    # print(f'width: {width}')
+    # print(f'height: {height}')
+    # print(f'position: {position}')
+    # print(f'position2: {position2}')
+
```
Comment (jourdain) [inline trame_vtk/modules/vtk/addon_serializer.py:200]: Should we remove all those commented lines?

File: trame_vtk/modules/vtk/addon_serializer.py [hunk 2]
```diff
@@ -64,6 +65,213 @@ def extractRequiredFields(
...
     # Override extractRequiredFields to fix handling of Normals/TCoords
     setattr(render_window_serializer, "extractRequiredFields", extractRequiredFields)
+    setattr(
+        render_window_serializer, "genericMapperSerializer", genericMapperSerializer
+    )
+    setattr(
+        render_window_serializer, "scalarBarActorSerializer", scalarBarActorSerializer
+    )
+
+    render_window_serializer.registerInstanceSerializer(
```
Comment (jourdain) [inline trame_vtk/modules/vtk/addon_serializer.py:256]: Maybe rework it like

```python
for name in ["vtkMapper", ...]:
   render_window_serializer.registerInstanceSerializer(name, genericMapperSerializer)
```

File: ignore-words.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+dragable
```
Comment (jourdain) [inline ignore-words.txt:1]: can we make that file private? `.ignore-words.txt`

Comment (banesullivan) [reply ignore-words.txt:1]: Will do

File: trame_vtk/modules/vtk/addon_serializer.py (hunk 1 above)
Comment (banesullivan) [reply trame_vtk/modules/vtk/addon_serializer.py:200]: I left to be consistent with the method it's patching...

I agree they should be removed. I'll clean this up here then after open a PR in VTK

**Comment by banesullivan** [comment]
@jourdain, this PR should be squash and merged and the PR title will be the commit message for semantic release

**Comment by jourdain** [comment]
Code looks good, you just need to force push your history rewrite.

**Comment by banesullivan** [comment]
Force pushing is really not ideal... but if I *must*. FWIW, just using squash and merge on GitHub for this PR would be nice and clean.

**Comment by banesullivan** [comment]
Force pushed

**Comment by jourdain** [comment]
I don't get `using squash and merge on GitHub for this PR would be nice and clean`. Do I need to enable something so I get the proper commit message from that squash? Also some PR should have several commits and kept that way...

**Comment by banesullivan** [comment]
Nothing needs to be enabled or disabled for the repo. When merging a PR, you select "squash and merge" which will then prompt you for the commit message (defaults to the PR title). 

This is significantly better than force-pushing and can be done on a per-PR basis so that you can squash when the changes went through iterations that you don't want in the history of the main branch or you can do a normal merge if wanting to capture that history.

Use the dropdown arrow next to the merge button to decide the type of merge:

Screen Shot 2022-10-20 at 10 17 27 AM


Then when you click the button, it prompts you:

Screen Shot 2022-10-20 at 10 17 51 AM

**Comment by jourdain** [comment]
Great, that means I can fix things myself. I never look closer in that drop down... Thx

**Pull request merged**